lib/iris/unit.py

```diff
@@ -1823,9 +1823,9 @@ def convert(self, value, other, ctype=FLOAT64):
             >>> import numpy as np
             >>> c = unit.Unit('deg_c')
             >>> f = unit.Unit('deg_f')
-            >>> print(c.convert(0, f))
+            >>> print(np.round(c.convert(0, f), 8))
```

Comment:
Yep, comes out the same:
```
$ source activate env27
$ python
>>> import iris.unit as unit
>>> c = unit.Unit('deg_c')
>>> f = unit.Unit('deg_f')
>>> c.convert(0, f)
31.999999999999886

$ source activate env34
$ python
>>> import iris.unit as unit
>>> c = unit.Unit('deg_c')
>>> f = unit.Unit('deg_f')
>>> c.convert(0, f)
31.999999999999886
```

If you're running the doctest, make sure you've got the patched one here (except for the rounding change, of course.)